e2e/e2e_test.go

```diff
@@ -87,17 +108,202 @@ var _ = Describe("Whereabouts functionality", func() {
+
+	By("Forcing reconciliation of the cluster...")
+	if err := runIpReconciler(kubeconfigPath); err != nil {
+		return err
+	}
+
+	By("Comparing pod ip assignments to ippool reservations")
+	// TODO - stop this ipPools[0] stuff and get it by name if there's a way to do it. low priority though
+	ipPools, err := k8sClient.ListIPPools(context.Background())
+	if err != nil {
+		return err
+	}
+
+	ipPoolAllocations := ipPools[0].Allocations() // todo, get this more elegantly than ipPools[0]
+	podList, err := ListPods(cs, testNamespace, wbLabelEqual)
+	if err != nil {
+		return err
+	}
+
+	found := true
+	podIP := ""
+	reservedIP := ""
+
+	for _, allocation := range ipPoolAllocations {
+		found = false
+		for _, pod := range podList.Items {
+			podIP, err = secondaryIfaceIPValue(&pod)
+			if err != nil {
+				return err
+			}
+			reservedIP = allocation.IP.String()
+
+			if reservedIP == podIP {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("Possible stale IP Pool: failed to find pod for IP Pool IP %s\n", allocation.IP)
+		}
+	}
+
+	for _, pod := range podList.Items {
+		found = false
+		for _, allocation := range ipPoolAllocations {
+			podIP, err = secondaryIfaceIPValue(&pod)
+			if err != nil {
+				return err
+			}
+			reservedIP = allocation.IP.String()
+
+			if reservedIP == podIP {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			fmt.Printf("possible pod IP not recorded in IP pool: failed to find IP pool allocation for pod IP %s\n", podIP)
+			return fmt.Errorf("possible pod IP not recorded in IP pool: failed to find IP pool allocation for pod IP %s\n", podIP)
+		}
+	}
+
+	return nil
+}
+
+type envVars struct {
+	kubeconfigPath      string
+	numComputeNodes     int
+	fillPercentCapacity int
+	numThrashIter       int
+}
+
+func getEnvVars() (envVars, error) {
 	const kubeconfig = "KUBECONFIG"
+	const numCompute = "NUMBER_OF_COMPUTE_NODES"
+	const fillCapcity = "FILL_PERCENT_CAPACITY"
+	const numThrash = "NUMBER_OF_THRASH_ITER"
+
+	var err error
+
+	envVars := envVars{}
 
 	kubeconfigPath, found := os.LookupEnv(kubeconfig)
 	if !found {
-		return nil, fmt.Errorf("must provide the path to the kubeconfig via the `KUBECONFIG` env variable")
+		return envVars, fmt.Errorf("must provide the path to the kubeconfig via the `KUBECONFIG` env variable")
+	}
+	numComputeNodes, found := os.LookupEnv(numCompute)
+	if !found {
+		return envVars, fmt.Errorf("must provide the number of compute nodes via the `NUMBER_OF_COMPUTE_NODES` env variable")
+	}
+	fillPercentCapacity, found := os.LookupEnv(fillCapcity)
+	if !found {
+		return envVars, fmt.Errorf("must provide the fill percent via the `FILL_PERCENT_CAPACITY` env variable")
+	}
+	numThrashIter, found := os.LookupEnv(numThrash)
+	if !found {
+		return envVars, fmt.Errorf("must provide the number of iterations via the `NUMBER_OF_THRASH_ITER` env variable")
 	}
```

Comment:
these have defaults on the bash script - I think we can preserve that behavior. 

I would probably use defaults more suitable to running this locally - 2 nodes, 50 percent capacity, 1 iteration - but I'm also OK if you prefer to re-use the existing defaults.

Reply:
I like the idea of default config being suited for local runs.